core/src/gl/shaderProgram.cpp

```diff
@@ -84,15 +120,20 @@ GLint ShaderProgram::getAttribLocation(const std::string& _attribName) {
 
 }
 
-GLint ShaderProgram::getUniformLocation(const std::string& _uniformName) {
+GLint ShaderProgram::getUniformLocation(const UniformEntries::UniformEntry* _entry) {
```

Comment:
I think the input here (and for the other `ShaderProgram` methods) could be just the ID of the uniform. When the program needs to look up the location on the first call, it can get the name using the uniform namespace functions - this reduces the indirection needed for all but the first call.